app/code/core/Maho/AdminActivityLog/etc/config.xml

```diff
@@ -0,0 +1,203 @@
+    </admin>
+    <adminhtml>
+        <menu>
+            <system>
+                <children>
+                    <adminactivitylog translate="title" module="adminactivitylog">
+                        <title>Admin Activity Log</title>
+                        <sort_order>100</sort_order>
+                        <children>
+                            <activity translate="title" module="adminactivitylog">
+                                <title>Activity Log</title>
+                                <action>adminhtml/adminactivitylog_activity</action>
+                                <sort_order>10</sort_order>
+                            </activity>
+                            <login translate="title" module="adminactivitylog">
+                                <title>Login Activity</title>
+                                <action>adminhtml/adminactivitylog_login</action>
+                                <sort_order>20</sort_order>
+                            </login>
+                        </children>
+                    </adminactivitylog>
+                </children>
+            </system>
+        </menu>
+        <acl>
+            <resources>
+                <admin>
+                    <children>
+                        <system>
+                            <children>
+                                <adminactivitylog translate="title" module="adminactivitylog">
+                                    <title>Admin Activity Log</title>
+                                    <sort_order>100</sort_order>
+                                    <children>
+                                        <activity translate="title">
+                                            <title>View Activity Log</title>
+                                        </activity>
+                                        <login translate="title">
+                                            <title>View Login Activity</title>
+                                        </login>
+                                    </children>
+                                </adminactivitylog>
+                            </children>
+                        </system>
+                    </children>
+                </admin>
+            </resources>
+        </acl>
+        <layout>
+            <updates>
+                <adminactivitylog>
+                    <file>adminactivitylog.xml</file>
+                </adminactivitylog>
+            </updates>
+        </layout>
+    </adminhtml>
+    <crontab>
+        <jobs>
+            <adminactivitylog_clean_old_logs>
+                <schedule>
+                    <cron_expr>0 2 * * *</cron_expr>
+                </schedule>
+                <run>
+                    <model>adminactivitylog/observer::cleanOldLogs</model>
+                </run>
+            </adminactivitylog_clean_old_logs>
+        </jobs>
+    </crontab>
+    <default>
+        <admin>
+            <adminactivitylog>
+                <enabled>1</enabled>
```

Comment:
Do you think it should be enabled by default? The table could potentially get pretty big and I assume only a subset of users will want this feature.

Reply:
My thought was, this is a feature that people wouldn't activate but then complain that it wasn't active when they need to search for something weird that happened on the DB. That's why I thought about enabling it although not everything is logged.

But also, there's only 30 days logged, the rest should be automatically removed

Reply:
I don't want to impose this so let's talk about it, I totally see your point

Reply:
I think it would be ok to only log the main action, but how to do that tho?

But actually order/shipping etc shouldn't even be logged because they're already `historical` in the sense that they've timestamps, they're not modifiable etc...

I think this PR is very useful for all of the data that it's not historically tracked (everything else actually, product change, inventory change, modifying a tax class, modifying a CMS block)

Reply:
> I think it would be ok to only log the main action, but how to do that tho?

The only thing I can think of is if we group it by "page visit". So you'd see something like `/catalog/product/save/id/123`, then click on that and see all the model saves that happened during that action. Not really sure the best way to implement that though, maybe using `preDispatch` somehow.